### ISSUE TITLE
Changing the name property in the upgrade callback creates type issues

### ISSUE THREAD

**Issue by WEBhicham** [jakearchibald/idb#116]
```ts
import { openDB, DBSchema, IDBPDatabase } from 'idb';
 
interface MyDBV1 extends DBSchema {
  'favourite-number': { key: string, value: number },
}
 
interface MyDBV2 extends DBSchema {
  'fave-num': { key: string, value: number },
}
 
const db = await openDB<MyDBV2>('my-db', 2, {
  async upgrade(db, oldVersion) {
    // Cast a reference of the database to the old schema.
    const v1Db = db as unknown as IDBPDatabase<MyDBV1>;
 
    if (oldVersion < 1) {
      v1Db.createObjectStore('favourite-number');
    }
    if (oldVersion < 2) {
      const store = v1Db.createObjectStore('favourite-number');
      store.name = 'fave-num';
    }
  }
});
```

This example doesn't work in practice. 
`error TS2322: Type '"fave-num"' is not assignable to type '"favourite-number"'.`

**Comment by jakearchibald** [comment]
Hmm. I'll have a think if there's a better way to handle this. As a workaround, cast to `IDBPDatabase` for the upgrade function.

**Comment by jakearchibald** [comment]
Fixed in 7b526a929939e6b41809e2d0429380dbe8fd131c
